projectroles/app_settings.py

```diff
@@ -260,6 +271,9 @@ def get_default(cls, app_name, setting_name, post_safe=False):
             app_settings = app_plugin.app_settings
 
         if setting_name in app_settings:
+            if callable(app_settings[setting_name].get('default')):
+                callable_setting = app_settings[setting_name].get('default')
+                return callable_setting(project, user)
```

Comment:
One thing I forgot to mention in my previous review: whenever we call these callable methods, we should do it as follows:

- Call within `try/catch` block
- IF exception is returned, log it with `logger.error()`
    * AND set default value into something appropriate for the setting type (e.g. empty string for string, 0 for int, False for boolean, {} for json. You should define these in a dict in the `app_settings` module, if we haven't done so already)
- Same as above for options, but simply return one option which is the default value of the setting.

This avoids a situation where a bug in the implementation in a single callable from an app crashes the project/user form entirely and prevents its usage.